hubstorage/resourcetype.py

```diff
@@ -18,6 +19,23 @@ def __init__(self, client, key, auth=None):
         self.auth = xauth(auth) or client.auth
         self.url = urlpathjoin(client.endpoint, self.key)
 
+    @property
+    def _supports_msgpack(self):
+        return self.resource_type in ['items', 'logs',
+                                      'collections', 'samples']
+
+    @staticmethod
+    def _enforce_msgpack(**kwargs):
+        kwargs['stream'] = True
```

Comment:
> I see that _iter_lines doesn't enforce stream=True, and iter_json below - does.

Right. Note that this is the current behavior in `master` [1]. IDK if this was introduced on purpose, but it looks like it's safe to enforce `stream=True` in `_iter_lines()`, since it's only used in `iter_json` (besides tests).

> Any reason why we don't follow this approach with msgpack?

Yes, `requests.Response.iter_content()`, used to download the msgpack bytes, requires `stream=True` to avoid reading everything at once [2].

[1] https://github.com/scrapinghub/python-hubstorage/blob/master/hubstorage/resourcetype.py#L76
[2] http://docs.python-requests.org/en/master/api/?highlight=iter_content#requests.Response.iter_content